src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/PackageModel.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.UI.Models.Package
+{
+    public abstract class PackageModel : IEmbeddedResourcesCapable
+    {
+        private readonly IEmbeddedResourcesCapable _embeddedResources;
+
+        internal PackageModel(PackageIdentity identity,
+            IEmbeddedResourcesCapable embeddedResources,
+            string? title = null,
+            string? description = null,
+            string? authors = null,
+            Uri? projectUrl = null,
+            string[]? tags = null,
+            IReadOnlyList<string>? ownersList = null,
+            IReadOnlyCollection<PackageDependencyGroup>? packageDependencyGroups = null,
+            string? summary = null,
+            DateTimeOffset? publishedDate = null,
+            LicenseMetadata? licenseMetadata = null,
+            Uri? licenseUrl = null,
+            bool requireLicenseAcceptance = false,
+            Uri? iconUrl = null)
+        {
+            _embeddedResources = embeddedResources ?? throw new ArgumentNullException(nameof(embeddedResources));
+            Identity = identity ?? throw new ArgumentNullException(nameof(identity));
+            Title = title;
+            Description = description;
+            Authors = authors;
+            ProjectUrl = projectUrl;
+            Tags = tags;
+            OwnersList = ownersList;
+            Summary = summary;
+            PublishedDate = publishedDate;
+            LicenseMetadata = licenseMetadata;
+            LicenseUrl = licenseUrl;
+            RequireLicenseAcceptance = requireLicenseAcceptance;
+            IconUrl = iconUrl;
+
+            if (packageDependencyGroups != null && packageDependencyGroups.Count > 0)
+            {
+                DependencySets = packageDependencyGroups.Select(e => new PackageDependencySetMetadata(e)).ToArray();
```

Comment:
```suggestion
                DependencySets = packageDependencyGroups.Select(e => new PackageDependencySetMetadata(e)).ToList();
```

`ToArray()` at best case will cause the same allocations as `ToList`, but worst case `ToArray` will create a List internally, and then allocate another array.

Basically, unless it's really needed, `ToArray` should be avoided.

Reply:
Fixed, thanks!